server/service/src/repack/insert.rs

```diff
@@ -0,0 +1,516 @@
+        let stock_line_repo = StockLineRowRepository::new(&connection);
+
+        // Repack increase where stock pack size == one
+        let increased_pack_size = service
+            .insert_repack(
+                &context,
+                inline_init(|r: &mut InsertRepack| {
+                    r.stock_line_id = mock_stock_line_a().id.clone();
+                    r.number_of_packs = 8.0;
+                    r.new_pack_size = 2;
+                }),
+            )
+            .unwrap();
+
+        let invoice = invoice_repo
+            .find_one_by_id(&increased_pack_size.invoice_row.id)
+            .unwrap();
+        let invoice_lines = invoice_line_repo
+            .find_many_by_invoice_id(&invoice.id)
+            .unwrap();
+
+        let stock_line_ids: Vec<String> = invoice_lines
+            .iter()
+            .map(|line| line.stock_line_id.clone().unwrap())
+            .collect();
+        let stock_lines = stock_line_repo.find_many_by_ids(&stock_line_ids).unwrap();
+
+        assert_eq!(invoice_lines.len(), 2);
+        // New stock line
+        assert_eq!(
+            stock_lines[0],
+            inline_init(|s: &mut StockLineRow| {
+                s.id = stock_lines[0].id.clone();
+                s.item_id = mock_stock_line_a().item_id.clone();
+                s.store_id = mock_stock_line_a().store_id.clone();
+                s.supplier_id = mock_stock_line_a().supplier_id.clone();
+                s.available_number_of_packs = 4.0;
+                s.total_number_of_packs = 4.0;
+                s.pack_size = 2;
+                s.cost_price_per_pack = mock_stock_line_a().cost_price_per_pack * 2.0;
+                s.sell_price_per_pack = mock_stock_line_a().sell_price_per_pack * 2.0;
+            })
+        );
+        assert_eq!(
+            stock_lines[1],
+            inline_edit(&mock_stock_line_a(), |mut s| {
+                s.available_number_of_packs = 22.0;
+                s.total_number_of_packs = 32.0;
+                s
+            })
+        );
+
+        // Repack increase where size != 1
+        let increased_pack_size = service
+            .insert_repack(
+                &context,
+                inline_init(|r: &mut InsertRepack| {
+                    r.stock_line_id = stock_line_a.id.clone();
+                    r.number_of_packs = 6.0;
+                    r.new_pack_size = 6;
+                }),
+            )
+            .unwrap();
+
+        let invoice = invoice_repo
+            .find_one_by_id(&increased_pack_size.invoice_row.id)
+            .unwrap();
+        let invoice_lines = invoice_line_repo
+            .find_many_by_invoice_id(&invoice.id)
+            .unwrap();
+
+        let stock_line_ids: Vec<String> = invoice_lines
+            .iter()
+            .map(|line| line.stock_line_id.clone().unwrap())
+            .collect();
+        let stock_lines = stock_line_repo.find_many_by_ids(&stock_line_ids).unwrap();
```

Comment:
order is not guaranteed here, in postgres test fails:

You can do something like this (may warrant a helper method in the test):

```rust
        let invoice_lines = InvoiceLineRepository::new(&connection)
            .query_by_filter(InvoiceLineFilter::by_invoice_id(&invoice.id))
            .unwrap();

        assert_eq!(invoice_lines.len(), 2);
        let (in_line, out_line) =
            if invoice_lines[0].invoice_line_row.r#type == InvoiceLineRowType::StockIn {
                (invoice_lines[0], invoice_lines[1])
            } else {
                (invoice_lines[1], invoice_lines[0])
            };

        let (updated_stock, new_stock) = (
            in_line.stock_line_option.clone().unwrap(),
            out_line.stock_line_option.clone().unwrap(),
        );
```

Reply:
oops `by_invoice_id` doesn't exist (was making an example and added it to my example for InvoiceLineFilter impl but forgot it's not here)